external-crates/move/crates/move-compiler/src/expansion/translate.rs

```diff
@@ -2578,10 +2617,10 @@ fn type_(context: &mut Context, sp!(loc, pt_): P::Type) -> E::Type {
             }
         }
         PT::Ref(mut_, inner) => ET::Ref(mut_, Box::new(type_(context, *inner))),
-        PT::Fun(_, _) => {
-            // TODO these will be used later by macros
-            context.spec_deprecated(loc, /* is_error */ true);
-            ET::UnresolvedError
+        PT::Fun(args, result) => {
```

Comment:
Should this be feature gated?

Reply:
This was present from specs too (though I did break the syntax... whoopsy (sorry not sorry (maybe a little sorry)))

Reply:
Yes, but feature gated to only expand / process lambdas if macros are enabled, and otherwise warn about deprecated specs.

Reply:
Let me take a look at this, and maybe just also try not to break people's specs (at least not for `Legacy` mode)